core/src/transaction.rs

```diff
@@ -141,91 +141,47 @@ impl CellOutput {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Debug, Default, OccupiedCapacity)]
+pub struct Witness(pub u32, pub Vec<Vec<u8>>);
+
+impl Witness {
+    // Index of relevant input
+    pub fn index(&self) -> u32 {
```

Comment:
Witness is not matched by position?

Reply:
>A non-witness program (defined hereinafter) txin MUST be associated with an empty witness field, represented by a 0x00

Some scripts don't need to use or just don't want to use witness.
Bitcoin enforce every txin associate with one witness(could by empty), and fixed the witness program format，such as

>* If the version byte is 0, and the witness program is 20 bytes:
The witness must consist of exactly 2 items (≤ 520 bytes each). The first one a signature, and the second one a public key.

It is not very suitable to port this rules.